auxx/src/Command/TyProjection.hs

```diff
@@ -0,0 +1,166 @@
+module Command.TyProjection
```

Comment:
This module has only one comment which is about some hack. Can you at least write a сomment to the whole module here?
I see that most of the definitions are trivial, but the what it all means is not clear for an unprepared reader.

Reply:
I must document other modules as well, but not in this sprint probably (writing good docs takes time). In short, this module is used for matching runtime values to Haskell types. If you have `ValueNumber 12.5`, can it be converted to an `Int`? (no, because of the fractional part). But `ValueNumber 4` can. And `TyProjection Int` is what does the conversion (it also contains the type name for an error message)